esmvaltool/diag_scripts/aerosols/aero_utils.py

```diff
@@ -0,0 +1,270 @@
+
+    Parameters
+    ----------
+    icube : Iris cube
+    pt_lat, pt_lon : latitude and longitude coordinates of desired points
+    keys: (optional)
+        height  : altitude (above geoid) of point
+        level   : model/ pseudo/ tile number, default = all
+        nearest : (optional) Whether to use 'nearest neighbour', instead
+                  of 'linear' method while extracting data.
+
+    Returns
+    -------
+    data_out : List
+        List of single point values, corresponding to each point specified.
+    """
+
+    # Check that input data is a (single) cube
+
+    if not isinstance(icube, iris.cube.Cube):
+        raise AeroAnsError('Extract_pt:First argument must be a single cube')
+
+    # Check that equal number of lat/lon pairs are passed in point coordinates
+    # convert arguments to lists as easier to process later
+    pt_lat1 = []
+    pt_lon1 = []
+
+    if not isinstance(pt_lat, list):
+        pt_lat1.append(pt_lat)
+        pt_lon1.append(pt_lon)
+
+    else:
+        for n in np.arange(len(pt_lat)):
+            pt_lat1.append(pt_lat[n])
+            pt_lon1.append(pt_lon[n])
+
+    if len(pt_lat1) != len(pt_lon1):
+        raise AeroAnsError('Extract_pt:Mismatch in number of lat/long values')
+
+    if 'level' in keys and 'height' in keys:
+        raise AeroAnsError('Extract_pt: Both Level and Height requested')
+
+    if icube.coords()[0].name() == 'time':  # only lat,lon,lev accepted
+        raise AeroAnsError(
+            'Extract_pt:Cannot handle time dimension at present')
+
+    if 'level' in keys and not icube.coord(
+            'model_level_number') and not icube.coord('pseudo_level'):
+        raise AeroAnsError('Extract_pt:Level requested, but not found in cube')
+
+    if 'height' in keys:
+        pt_hgt = []
+
+        if not isinstance(keys['height'], list):
+            pt_hgt.append(keys['height'])
+
+        else:
+            for n in np.arange(len(keys['height'])):
+                pt_hgt.append(keys['height'][n])
+
+        if len(pt_lat1) != len(pt_hgt):
+            raise AeroAnsError(
+                'Extract_pt:Mismatch in number of points for lat/long/height')
+
+        # Check that heights have been merged with orography.
+        if not icube.coords('altitude'):
+            raise AeroAnsError(
+                'Extract_pt:Height requested but input data does not contain \
+                "Altitude" coordinate')
+        else:
+
+            # Store the min and max altitudes from cube data so that user
+            # cannot request points located below/ above that.
+            # Will extract =min/max if beyond limits.
+            hgt_min = icube.coord('altitude').points.min()
+            hgt_max = icube.coord('altitude').points.max()
+
+    # ---------- Finished checks -- begin processing -------------------------
+
+    # If level specified, extract slice first
+    if 'level' in keys:
+
+        if icube.coord('model_level_number'):
+            icube = icube.extract(
```

Comment:
these extractions are always better off inside a try/except block (unless you've already checked for that var in the previous checks, that I may have missed)